src/ploomber_core/config.py

```diff
@@ -112,19 +116,46 @@ def __setattr__(self, name, value):
             raise ValueError(f"{name} not a valid field")
         else:
             super().__setattr__(name, value)
-            self._write()
+
+            # Check if the filesystem is writable
+            if self.filesystem_writable() and name != "writable":
+                self._write()
 
     def load_config(self):
         config = None
-        path = self.path()
-        is_config_exist = Path.is_file(path)
-        if is_config_exist:
-            text = path.read_text()
-            if text:
-                config = yaml.safe_load(text)
+
+        if self.writable:
+            path = self.path()
+            is_config_exist = Path.is_file(path)
+            if is_config_exist:
+                text = path.read_text()
+                if text:
+                    config = yaml.safe_load(text)
 
         return config
 
+    def random_string(self, length=10):
+        """Generate a random string
+
+        Parameters
+        ----------
+        length
+            Length of the string
+        """
+        sequence = f"{string.ascii_uppercase}{string.digits}"
+        return "".join(random.choices(sequence, k=length))
+
+    def filesystem_writable(self):
+        """Check if the filesystem is writable"""
+        try:
+            # random string is used for race conditions when using multiprocessing
+            tmp = self.path().parent / f"tmp_{self.random_string()}.txt"
+            tmp.touch()
+            tmp.unlink()
+            return True
+        except PermissionError:
+            return False
```

Comment:
it's a best practice to have a single line in the try statement:



```python
        try:
            tmp.touch()
            return True
        except PermissionError:
            return False
```

you can move the other lines outside the try statement.

and you can move the unlink in the `else` statement (after the except statement)